trustfall_core/src/ir/mod.rs

```diff
@@ -1,3 +1,4 @@
+//! Trustfall internal representation (IR)
```

Comment:
```suggestion
//! Trustfall intermediate representation (IR)
```
Though it can stand for both meanings for sure. I prefer "intermediate" slightly because it doesn't give the accidental impression that the IR is meant to only be readable inside Trustfall itself, which is counter to the vision of having a universal data layer: "LLVM for data sources."